subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -376,19 +467,44 @@ int nrf_cloud_obj_num_add(struct nrf_cloud_obj *const obj, const char *const key
 int nrf_cloud_obj_str_add(struct nrf_cloud_obj *const obj, const char *const key,
 			  const char *const val, const bool data_child)
 {
-	if (!obj || !key || !val) {
+	if (!obj || !val) {
 		return -EINVAL;
 	}
 
 	switch (obj->type) {
 	case NRF_CLOUD_OBJ_TYPE_JSON:
 	{
+		if (!key) {
+			return -EINVAL;
+		}
+
 		if (!obj->json) {
 			return -ENOENT;
 		}
 		return cJSON_AddStringToObjectCS(dest_json_get(obj, data_child),
 						 key, val) ? 0 : -ENOMEM;
 	}
+	case NRF_CLOUD_OBJ_TYPE_COAP_CBOR:
+	{
+		if (!obj->coap_cbor) {
+			return -ENOENT;
+		}
+
+		if (obj->coap_cbor->str_val) {
```

Comment:
instead, check the union type to see if it has already been set:
e.g.: `if (obj->coap_cbor->type != NRF_CLOUD_DATA_TYPE_NONE)`